### PR TITLE
Support more request options

### PR DESCRIPTION
**Pull request by hfcorriez**
Some RSS need more request options to fetch, such as:

- Apple RSS need normal browser UA
- Private RSS need auth 
- Intern RSS need proxy
- and so on...

So I extend req as request options to handle more

### CODE REVIEW

**Comment by hfcorriez** [comment]
Any suggestion?

File: lib/find-rss.js [hunk 1]
```diff
@@ -33,6 +39,7 @@
           if (err) {
             return cb(err);
           }
+          console.log(html);
```
Comment (nikezono) [inline lib/find-rss.js]: please delete `console.log` for debug.

File: lib/find-rss.js [hunk 2]
```diff
@@ -142,11 +150,9 @@
     }
   };
 
-  requestAndEncodeWithDetectCharset = function(url, callback) {
-    return request.get({
-      uri: url,
-      encoding: null
-    }, function(err, res, body) {
+  requestAndEncodeWithDetectCharset = function(req, callback) {
+    console.log(req);
```
Comment (nikezono) [inline lib/find-rss.js]: same above.

**Comment by nikezono** [comment]
thanks for you PR, and it's looks good.
LGTM without https://github.com/nikezono/node-find-rss/pull/22#discussion_r36757012.

**Comment by hfcorriez** [comment]
Thanks, I had resolved them.

**Comment by nikezono** [comment]
fantastic :wolf:

**Comment by hfcorriez** [comment]
Expect your npm update

**Comment by nikezono** [comment]
updated.

**Comment by hfcorriez** [comment]
Good work. thx.

**Pull request merged**